src/suggest/date-suggest.ts

```diff
@@ -104,19 +125,76 @@ export default class DateSuggest extends CodeMirrorSuggest<IDateCompletion> {
   ): void {
     const includeAlias = event.shiftKey;
 
-    const head = this.getStartPos();
-    const anchor = this.cmEditor.getCursor();
+    const { workspace } = this.app;
+    const activeView = workspace.getActiveViewOfType(MarkdownView);
+
+    if (!activeView) {
+      return;
+    }
+    const editor = activeView.editor;
+    const head = this.startPos;
+    const anchor = editor.getCursor();
+    let dateStr = "";
+    let makeIntoLink = this.plugin.settings.autosuggestToggleLink;
+
+    if (suggestion.label.startsWith("time:")) {
+      const timePart = suggestion.label.substring(5);
+      dateStr = this.plugin.parseTime(timePart).formattedString;
+      makeIntoLink = false;
+    } else {
+      dateStr = this.plugin.parseDate(suggestion.label).formattedString;
+    }
 
-    let dateStr = this.plugin.parseDate(suggestion.label).formattedString;
-    if (this.plugin.settings.autosuggestToggleLink) {
+    if (makeIntoLink) {
       if (includeAlias) {
         dateStr = `[[${dateStr}|${suggestion.label}]]`;
       } else {
         dateStr = `[[${dateStr}]]`;
       }
     }
 
-    this.cmEditor.replaceRange(dateStr, head, anchor);
+    editor.replaceRange(dateStr, head, anchor);
     this.close();
   }
+
+  onTrigger(
+    cursor: EditorPosition,
+    editor: Editor,
+    file: TFile
+  ): EditorSuggestTriggerInfo {
+    const lineContents = editor.getLine(cursor.line);
+
+    const match = lineContents
```

Comment:
I think I need to play around with this autosuggest class a little bit myself since this doesn't seem like the easiest way to handle things. It's weird that you need to save a ref to the code mirror instance and that you need to reread the line each time. If the autosuggest is open, is this method still called? If so you can probably avoid reading the line contents and just seeing if there is already an open autosuggest (by seeing if `this.startPos` exists.